### PR TITLE
#37 showcase nextboolean

### LINKED ISSUE #37: Showcase nextBoolean
Add a Schowcase in Random with NeodymiumRandom.nextBoolean(int trueCaseProbability)

### PR DESCRIPTION
**Pull request by ufjena**

### CODE REVIEW

**Comment by ufjena** [comment]
@georgkunze please review the RandomProbabilityTest class.
Please also have a look at the Javadoc in all random test classes.

**Comment by georgkunze** [comment]
@ufjena the logmessages are missleading and are not in all cases represent the actual percentage entered/observable. Could you look into that?

File: src/test/java/showcase/neodymium/tests/pdf/PdfTextContentCheckTest.java [hunk 1]
```diff
@@ -23,11 +23,6 @@
  * implemented in Java and the execution environment is configured with Maven any available 3rd party technology can be
  * added to the test automation.<br>
  * This show case gives an basic example how to implement an PDF content check using pdfbox.<br>
- * <br>
```
Comment (occupant23) [inline src/test/java/showcase/neodymium/tests/pdf/PdfTextContentCheckTest.java:26]: Please add this as information to documentation, that the dependency is required but already prepared by us.

File: src/test/java/showcase/neodymium/tests/random/FixedRandomTest.java [hunk 2]
```diff
@@ -14,6 +14,18 @@
...
 
+/**
+ * This show case shows the use of the Neodymium.getRandom() method.<br>
+ * Random values are useful to run different test cases with different values.<br>
+ * In some cases it is necessary to be able to repeat a test with the same random value (e.g. debugging).<br>
+ * Here you can see how a fixed random value can be used.<br>
+ * <br>
+ * <b>REQUIRED CONFIGURATION:</b> <i>config/neodymium.properties</i>
+ * <ul>
+ * <li>A value for neodymium.context.random.initialValue must be set.</li>
```
Comment (occupant23) [inline src/test/java/showcase/neodymium/tests/random/FixedRandomTest.java]: you can already state the expected value in the documentation  

`... must be set to 123456789.`

File: src/test/java/showcase/neodymium/tests/random/RandomProbabilityTest.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+ * The NeodymiumRandom class extends the functionality of java.util.Random.<br>
+ * In this case the use of <i>nextBoolean(trueCaseProbability)</i> is shown.<br>
+ * With trueCaseProbability you can determine how often true or false is returned.<br>
+ * Please note that the result can only be an approximation of the specified percentage.<br>
+ * To get to know more possibilities take a closer look at the class.<br>
+ */
+@SuppressBrowsers
+public class RandomProbabilityTest extends AbstractTest
+{
+    private static Logger logger = LogManager.getLogger(LoggingFirstTest.class);
```
Comment (occupant23) [inline src/test/java/showcase/neodymium/tests/random/RandomProbabilityTest.java]: better 
`private static final Logger LOGGER`

File: src/test/java/showcase/neodymium/tests/random/RandomProbabilityTest.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            final boolean randomBoolean = NeodymiumRandom.nextBoolean(0);
+            if (randomBoolean)
+                trueCount++;
+            if (!randomBoolean)
+                falseCount++;
+        }
+        logger.error("Count of true:  " + trueCount + "    Count of false:  " + falseCount);
+
+        // Part 2: create a random boolean with probability 50
+        // this means that true and false are returned in equal parts
```
Comment (occupant23) [inline src/test/java/showcase/neodymium/tests/random/RandomProbabilityTest.java]: this mean that true and false should be returned approximately equally probable.

File: src/test/java/showcase/neodymium/tests/random/RandomProbabilityTest.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+            final boolean randomBoolean = NeodymiumRandom.nextBoolean(50);
+            if (randomBoolean)
+                trueCount++;
+            if (!randomBoolean)
+                falseCount++;
+        }
+        logger.error("Count of true:  " + trueCount + "    Count of false:  " + falseCount);
+
+        // Part 3: create a random boolean with probability 80
+        // this means that true is returned at 80% and false at 20%
```
Comment (occupant23) [inline src/test/java/showcase/neodymium/tests/random/RandomProbabilityTest.java]: this means the ratio of true and false should be approximately about 80 to 20 in favor of true.

**Review by occupant23 — CHANGES_REQUESTED** [review]
please see the comments

File: src/test/java/showcase/neodymium/tests/logging/LoggingFirstTest.java [hunk 6]
```diff
@@ -9,17 +9,18 @@
 import io.qameta.allure.Description;
 import showcase.neodymium.tests.AbstractTest;
 
+/**
+ * Neodymium provides extended logging. The log4j-api and log4j-core are included in the Neodymium library.<br>
+ * Therefore it is not necessary to add an entry in the pom.<br>
+ * ConfigurationFactory class will look for <i>log4j2-test.properties</i> in the classpath.<br>
+ * Then it will look for log4j2.properties on the classpath. The same goes for yaml, json and xml files.<br>
```
Comment (occupant23) [inline src/test/java/showcase/neodymium/tests/logging/LoggingFirstTest.java]: The same mechanism applies for other file formats like yaml, json and xml.

File: src/test/java/showcase/neodymium/tests/pdf/PdfTextContentCheckTest.java (hunk 1 above)
Comment (occupant23) [reply src/test/java/showcase/neodymium/tests/pdf/PdfTextContentCheckTest.java:26]: @ufjena this hasn't been covered yet.

File: src/test/java/showcase/neodymium/tests/random/RandomProbabilityTest.java [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+            final boolean randomBoolean = NeodymiumRandom.nextBoolean(0);
+            if (randomBoolean)
+                trueCount++;
+            if (!randomBoolean)
+                falseCount++;
+        }
+        logger.error("Count of true:  " + trueCount + "    Count of false:  " + falseCount);
+
+        // Part 2: create a random boolean with probability 50
+        // this mean that true and false should be returned approximately equally probable
```
Comment (occupant23) [inline src/test/java/showcase/neodymium/tests/random/RandomProbabilityTest.java]: this means that true and false should be returned approximately equally

**Review by occupant23 — CHANGES_REQUESTED** [review]

File: src/test/java/showcase/neodymium/tests/pdf/PdfTextContentCheckTest.java [hunk 8]
```diff
@@ -23,11 +23,8 @@
...
- * <b>REQUIRED CONFIGURATION</b> <i>pom.xml</i>:
- * <ul>
- * <li>org.apache.pdfbox/pdfbox must be added to the dependencies section</li>
- * </ul>
+ * In order to be able to use pdfbox, a dependency entry <i>org.apache.pdfbox</i> is required in the <i>pom.xml but
```
Comment (occupant23) [inline src/test/java/showcase/neodymium/tests/pdf/PdfTextContentCheckTest.java]: In order to use PDFbox you would need to add a dependency for <i>org.apache.pdfbox</i> to your project's <i>pom.xml</i> file. For this show case project we already prepared this and you can have a look if you're struggling with the setup.<br>

**Review by occupant23 — CHANGES_REQUESTED** [review]
On last thing :-)

**Review by occupant23 — APPROVED** [review]

**Pull request merged**